src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```
@@ -152,17 +160,74 @@ private void switchTimelines(final EventType eventType,
     }
 
+    public void delete(final String eventTypeName, final String timelineId, final Client client)
+            throws ForbiddenAccessException, UnableProcessException, TimelineException {
+        if (!client.getClientId().equals(securitySettings.getAdminClientId())) {
+            throw new ForbiddenAccessException("Request is forbidden for user " + client.getClientId());
+        }
+
+        final UUID uuid = uuidGenerator.fromString(timelineId);
+        final List<Timeline> timelines = timelineDbRepository.listTimelines(eventTypeName);
+        if (timelines.size() != DELETE_ONLY_FIRST_TIMELINE) {
+            throw new UnableProcessException("Timeline with timelineId: " + uuid + " could not be deleted");
+        }
+
+        final Timeline activeTimeline = timelines.get(0);
+        if (activeTimeline.getId().equals(uuid)) {
+            switchToFakeTimeline(uuid, activeTimeline);
+        } else {
+            throw new UnableProcessException("Timeline with timelineId: " + uuid + " not found");
+        }
+    }
+
+    private void switchToFakeTimeline(final UUID uuid, final Timeline activeTimeline) throws TimelineException {
+        LOG.info("Reverting timelines from {} to fake timeline", activeTimeline);
+        try {
+            timelineSync.startTimelineUpdate(activeTimeline.getEventType(), nakadiSettings.getTimelineWaitTimeoutMs());
+            timelineDbRepository.deleteTimeline(uuid);
+        } catch (final InterruptedException ie) {
+            Thread.currentThread().interrupt();
+            throw new TimelineException("Failed to switch timeline for:" + activeTimeline.getEventType());
+        } finally {
+            try {
+                timelineSync.finishTimelineUpdate(activeTimeline.getEventType());
+            } catch (final InterruptedException ie) {
+                Thread.currentThread().interrupt();
+                throw new TimelineException("Timeline update was interrupted for:" + activeTimeline.getEventType());
```

This may eat original exception